runtime/lua/vim/lsp/handlers.lua

```diff
@@ -451,9 +457,12 @@ end
 
 -- Add boilerplate error validation and logging for all of these.
 for k, fn in pairs(M) do
-  M[k] = function(err, method, params, client_id, bufnr, config)
+  M[k] = function(err, result, ctx, config)
+    local client_id = ctx.client_id
+    local method = ctx.method
+    local bufnr = ctx.bufnr
     local _ = log.debug() and log.debug('default_handler', method, {
-      params = params, client_id = client_id, err = err, bufnr = bufnr, config = config
+      result = result, client_id = client_id, err = err, bufnr = bufnr, config = config
     })
```

Comment:
Might be worth logging the context with vim.inspect, to ensure this automatically logs everything if the context gets extended.

Reply:
vim.inspect has a runtime cost though right? I guess it's fine since this is only done if the user is using debug

Reply:
I'd think the overhead is acceptable given that it is logged on debug level and that is not enabled by default.